epam/edp-tekton
# chore: Make Tekton Dashboard quickLink deployment optional(#246)

Description
Implemented conditional logic for deploying the Tekton Dashboard QuickLink in the edp-tekton Helm chart. This change makes the deployment of the Tekton Dashboard QuickLink optional, based on the deployment status of the Tekton Dashboard itself. The adjustment ensures greater flexibility and customization for users managing their Tekton resources.

Fixes #246

Type of change
How Has This Been Tested?

Deployed the Helm chart with the Tekton Dashboard enabled to verify that the QuickLink gets deployed.
Deployed the Helm chart with the Tekton Dashboard disabled to confirm that the QuickLink does not get deployed.
Ensured that existing functionalities are not affected by performing regression testing around the Tekton resources deployment.

Instructions for Reproduction:

- Set dashboard.enabled to true in the values file and deploy. Verify that the QuickLink is deployed.
- Set dashboard.enabled to false in the values file and redeploy. Check that the QuickLink is not deployed.

Checklist:
- [ ] I have made corresponding changes to the documentation reflecting the optional deployment feature.
- [x] The Pull Request contains a single commit, which has been squashed from multiple commits addressing the task.
Additional context
This change was driven by the need for configurable deployment options, allowing users to tailor the Tekton resources deployment to their specific requirements. This enhancement promotes modularity and adaptability in resource management within the Tekton ecosystem.
